demux/demux_lavf.c

```diff
@@ -647,6 +647,19 @@ static int dict_get_decimal(AVDictionary *dict, const char *entry, int def)
     return def;
 }
 
+static bool is_image(int64_t nb_frames, bool attached_picture,
+                     const char *avifname, enum AVCodecID codec_id)
+{
+    return nb_frames <= 1 && (
+        attached_picture ||
+        bstr_endswith0(bstr0(avifname), "_pipe") ||
+        strcmp(avifname, "alias_pix") == 0 ||
+        strcmp(avifname, "gif") == 0 ||
+        strcmp(avifname, "image2pipe") == 0 ||
+        (codec_id == AV_CODEC_ID_AV1 && nb_frames == 1)
+    );
```

Comment:
if you pass the entire `AVStream` you won't need to pass codec_id separately
also I'd pass the `AVInputFormat` (instead of `avifname`) for cleanliness reasons